File: specs/esm/overview-detail.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadParquet("walk", "data/random-walk.parquet")
]);

const $brush = vg.Selection.intersect();

export default vg.vconcat(
  vg.plot(
    vg.areaY(
      vg.from("walk"),
      {x: "t", y: "v", fill: "steelblue"}
    ),
    vg.intervalX({as: $brush}),
    vg.width(680),
    vg.height(200)
  ),
  vg.plot(
    vg.areaY(
      vg.from("walk", {filterBy: $brush}),
      {x: "t", y: "v", fill: "steelblue"}
    ),
    vg.yDomain(vg.Fixed),
    vg.width(680),
    vg.height(200)
  )
);

File: docs/public/specs/python/overview-detail.py
import vgplot as vg

walk = vg.parquet("data/random-walk.parquet")

brush = vg.selection.intersect()

view = vg.vconcat(
    vg.plot(
        vg.area_y(walk, x="t", y="v", fill="steelblue"),
        vg.interval_x(bind=brush),
        vg.width(680),
        vg.height(200),
    ),
    vg.plot(
        vg.area_y(data=walk, filter_by=brush, x="t", y="v", fill="steelblue"),
        vg.y_domain("Fixed"),
        vg.width(680),
        vg.height(200),
    ),
)
